### ISSUE TITLE
p-dialog Accessibility Issue

### ISSUE THREAD

**Issue by aciesla4** [primefaces/primeng#18605]
### Describe the bug

The p-dialog component has an accessibility violation when running the IBM Equal Accessibility Checker: A hidden element should not contain any tabbable elements. 

### Reason for not contributing a PR

- [x] Lack of time
- [ ] Unsure how to implement the fix/feature
- [ ] Difficulty understanding the codebase
- [ ] Other

### Reproducer

https://fsgqoqui-foxruyzf.stackblitz.io

### Environment

Stackblitz

### Angular version

20.0.4

### PrimeNG version

v20

### Steps to reproduce the behavior

1. Install the [IBM Equal Accessibility Checker](https://chromewebstore.google.com/detail/ibm-equal-access-accessib/lkcagbfjnkomcinoddgooolagloogehp)
2. Open the stackblitz
3. Run the accessibility checker
4. Filter on violations only and note the error: 'Element "span" should not be focusable within the subtree of an element with an 'aria-hidden' attribute with value 'true'

### Expected behavior

No accessibility violations are present for the p-dialog component
